homeassistant/components/kostal_plenticore/helper.py

```diff
@@ -182,9 +187,137 @@ async def _async_update_data(self) -> dict[str, dict[str, str]]:
         _LOGGER.debug("Fetching %s for %s", self.name, self._fetch)
 
         fetched_data = await client.get_setting_values(self._fetch)
+        return fetched_data
+
+    async def async_read_data(self, module_id: str, data_id: str) -> [str, bool]:
+        """Writes settings back to Plenticore."""
+        client = self._plenticore.client
+
+        if not self._fetch or client is None:
+            return False
+
+        try:
+            val = await client.get_setting_values(module_id, data_id)
+        except PlenticoreApiException:
+            return False
+        else:
+            return val
+
+    async def async_write_data(self, module_id: str, value: dict[str, str]) -> bool:
+        """Writes settings back to Plenticore."""
+        client = self._plenticore.client
+
+        if not self._fetch or client is None:
+            return False
+
+        try:
+            await client.set_setting_values(module_id, value)
+        except PlenticoreApiException:
+            return False
+        else:
+            return True
+
+
+class PlenticoreSelectUpdateCoordinator(DataUpdateCoordinator):
```

Comment:
What's the point of this class?

Reply:
PlenticoreSelectUpdateCoordinator was needed to pass the options for the select forward.
This is not possible with PlenticoreUpdateCoordinator.